modules/Attendance/Services/AttendanceService.php

```
@@ -543,7 +543,15 @@ public function getTeamAttendance(string $requestingUserId, array $filters, ?int
         $attendanceByUserAndDate = [];
         foreach ($realAttendanceRecords as $record) {
             $userId = (string)$record->user_id;
-            $date = $record->clock_in_time->copy()->setTimezone($timezone)->format('Y-m-d');
+            // Ensure clock_in_time is a Carbon object before calling methods on it
+            if ($record->clock_in_time instanceof Carbon) {
+                $date = $record->clock_in_time->copy()->setTimezone($timezone)->format('Y-m-d');
+            } else if (is_string($record->clock_in_time)) {
+                $date = Carbon::parse($record->clock_in_time, $timezone)->format('Y-m-d');
+            } else {
+                // Default fallback if clock_in_time is null or invalid
+                $date = Carbon::now($timezone)->format('Y-m-d');
```

Using current date as fallback might mask data issues. Consider throwing an exception or logging a warning when clock_in_time is invalid/null to make debugging easier.